core/lib/zksync_core/src/state_keeper/updates/miniblock_updates.rs

```diff
@@ -130,6 +132,8 @@ impl MiniblockUpdates {
         self.l1_gas_count += tx_l1_gas_this_tx;
         self.block_execution_metrics += execution_metrics;
         self.txs_encoding_size += tx.bootloader_encoding_size();
+        self.payload_encoding_size +=
+            zksync_protobuf::repr::encode::<zksync_dal::models::proto::Transaction>(&tx).len();
```

Comment:
Dumb question: All executed transactions are preserved in `executed_transactions` (`TransactionExecutionResult` has a `transaction: Transaction` field). Wouldn't it be possible to reconstruct the payload size based on this info instead of introducing an extra field? Or would that be too inefficient?

Reply:
Yes, it's possible, but inefficient because it will require to encode all transactions each time, instead of just the appended transaction. So it's just a typical computation-storage tradeoff.